### ISSUE TITLE
 Performance issues

### ISSUE THREAD

**Issue by juzhen21510** [Ravenwall/node-fixed-array#2]
if push items in the fixed-array too fast, the array object will frequently shift，the peformance will be very low?

**Comment by brycebaril** [comment]
@juzhen21510 yeah, this is true. It could be possible to do some batch removal of items over the size limit, though it would still require some changes.

Something like: during `push`, instead of doing the `splice` and immediately updating the sum, move the `sum` property/calculation to a method and move the `splice` to every "read" method. It would still have to periodically do the `splice` even without any "read" commands being called to prevent memory bloat, though.
